lasagne/layers/recurrent.py

```diff
@@ -733,11 +780,16 @@ def step_masked(input_dot_W_n, mask_n, cell_previous, hid_previous,
             hid_init = T.dot(ones, self.hid_init)  # repeat num_batch times
 
         non_seqs = [self.W_hid_stacked]
-
         if self.peepholes:
             non_seqs += [self.W_cell_to_ingate,
                          self.W_cell_to_forgetgate,
                          self.W_cell_to_outgate]
+        else:
+            non_seqs += [(), (), ()]
```

Comment:
Im not inside theano :) But it raises the error

```
raise AsTensorError("Cannot convert %s to TensorType" % str_x, type(x))
theano.tensor.var.AsTensorError: ('Cannot convert None to TensorType', <type 'NoneType'>)
```
If the lists contrain `None` instead of `()`. 

I forgot to remove the None's from step. I need to fill up with empty tuples beacuse if peepholes is false and precompute is false then the first 3 arguments should be empty and the last two arguments should be [W_in_stacked, b_stacked].

I think for GRU we could omit the empty tuples but i included it for consistency.